assets/js/blocks/cart-checkout-shared/payment-methods/payment-method-error-boundary.tsx

```diff
@@ -2,51 +2,77 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { useState } from '@wordpress/element';
+import { Component } from '@wordpress/element';
 import { CURRENT_USER_IS_ADMIN } from '@woocommerce/settings';
 import { StoreNoticesContainer } from '@woocommerce/blocks-components';
 import { noticeContexts } from '@woocommerce/base-context';
 import { NoticeType } from '@woocommerce/types';
+import {
+	DerivedStateReturn,
+	ReactError,
+} from '@woocommerce/base-components/block-error-boundary/types';
+import { __experimentalDeRegisterExpressPaymentMethod } from '@woocommerce/blocks-registry';
+
 interface PaymentMethodErrorBoundaryProps {
 	isEditor: boolean;
 	children: React.ReactNode;
+	expressPaymentMethodId: string | undefined;
 }
-const PaymentMethodErrorBoundary = ( {
-	isEditor,
-	children,
-}: PaymentMethodErrorBoundaryProps ) => {
-	const [ errorMessage ] = useState( '' );
-	const [ hasError ] = useState( false );
-	if ( hasError ) {
-		let errorText = __(
-			'We are experiencing difficulties with this payment method. Please contact us for assistance.',
-			'woo-gutenberg-products-block'
-		);
-		if ( isEditor || CURRENT_USER_IS_ADMIN ) {
-			if ( errorMessage ) {
-				errorText = errorMessage;
-			} else {
-				errorText = __(
-					"There was an error with this payment method. Please verify it's configured correctly.",
-					'woo-gutenberg-products-block'
+
+class PaymentMethodErrorBoundary extends Component< PaymentMethodErrorBoundaryProps > {
+	state = { errorMessage: '', hasError: false };
+	expressPaymentMethodId = this.props.expressPaymentMethodId;
+	static getDerivedStateFromError( error: ReactError ): DerivedStateReturn {
+		return {
+			errorMessage: error.message,
+			hasError: true,
+		};
+	}
+
+	render() {
+		const { hasError, errorMessage } = this.state;
+		const { isEditor } = this.props;
+
+		if ( hasError ) {
+			let errorText = __(
+				'We are experiencing difficulties with this payment method. Please contact us for assistance.',
+				'woo-gutenberg-products-block'
+			);
+			if ( isEditor || CURRENT_USER_IS_ADMIN ) {
+				if ( errorMessage ) {
+					errorText = errorMessage;
+				} else {
+					errorText = __(
+						"There was an error with this payment method. Please verify it's configured correctly.",
+						'woo-gutenberg-products-block'
+					);
+				}
+			}
+			const notices: NoticeType[] = [
+				{
+					id: '0',
+					content: errorText,
+					isDismissible: false,
+					status: 'error',
+				},
+			];
+
+			// De-register the express payment method if has an error.
+			if ( this.expressPaymentMethodId !== undefined ) {
+				__experimentalDeRegisterExpressPaymentMethod(
+					this.expressPaymentMethodId
```

Comment:
Such calls shouldn't be done in the `render()` method, but in effects, i.e., the lifecycle methods like `componentDidMount` and `componentDidUpdate`.

Also, is it a good idea to unregister the method? Maybe the `PaymentMethodErrorBoundary` should do just one job, reporting the error, and nothing else.